src/screens/CreatePost/elements/SheetAddTopic.js

```diff
@@ -143,9 +143,9 @@ const SheetAddTopic = ({refTopic, onAdd, topics, onClose, chatTopics}) => {
                     if (v.match(/\s+$/gm)) {
                       return add();
                     }
-                    setTopic(v);
+                    setTopic(v?.toLowerCase());
                     if (v !== '') {
-                      searchTopic(v);
+                      searchTopic(v?.toLowerCase());
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are good for ensuring case-insensitive search. However, it's important to consider that the `toLowerCase()` function can throw a TypeError if `v` is undefined. To prevent this, you should check if `v` is defined before calling `toLowerCase()`. 

```diff
- setTopic(v?.toLowerCase());
+ setTopic(v ? v.toLowerCase() : '');
```
```diff
- searchTopic(v?.toLowerCase());
+ searchTopic(v ? v.toLowerCase() : '');
```
This way, if `v` is undefined, an empty string will be used instead, preventing any potential errors.


<!-- This is an auto-generated comment by OSS CodeRabbit -->